server/routes/induction/common/additionalTrainingController.ts

```diff
@@ -20,8 +20,7 @@ export default abstract class AdditionalTrainingController extends InductionCont
 
     // Check if we are in the midst of changing the main induction question set (e.g. from long route to short route)
     if (req.session.updateInductionQuestionSet) {
-      const { prisonNumber } = req.params
-      this.addCurrentPageToHistory(req, `/prisoners/${prisonNumber}/induction/additional-training`)
+      this.addCurrentPageToHistory(req)
```

Comment:
I have refactored the method `addCurrentPageToHistory` because it was kinda flawed previously because you had to pass in the page URL as a method arg. This has 2 problems - firstly you could pass in any page URL which might not necessarily be the "current" page; and secondly (more importantly) when this method is called in the superclass like this (which is used for both the create and update journeys), then the URL can only reflect one of those journeys.

I've refactored the method so that it only takes the `request` as the method arg, and it takes the current page from `req.path`

Because I've refactored this method, there is this change to pretty much all of the superclasses that call this method, and also the unit tests (because they now need to correctly setup `req.path` in the mock)